app/src/main/java/com/example/bottomnav/DoubleDecoder.java

```diff
@@ -0,0 +1,17 @@
+package com.example.bottomnav;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+public class DoubleDecoder<T> extends IntegerDecoder {
```

Comment:
I don't think this should extend IntegerDecoder. Extending (subclassing) means an is-a relation, which this is not. This uses some shared methods, but it fundamental is not an integer decoder. You might consider having an intermediate class in the hierarchy, eg PrimitiveDecoder, that is defined as taking the next n contiguous bytes, maybe decoding them as some format, then some some abstract transformation. Both Float, Double, and IntegerDecoders would implement that type.

For example:
```java
class PrimitveDecoder extends DataDecoder {
  public PrimitiveDecoder(int byteSize, String dataType) {
    ...
  }
  String decode(...) {
    ByteBuffer bytes = ...;  // this just gets the next n bytes
    transform(bytes)  // then leaves it to the subclass to do something with them
  }
  
  public T transform(ByteBuffer bytes)  // to be implemented by subclass
}
```